esp/esp/program/modules/base.py

```diff
@@ -510,16 +513,23 @@ def _checkGrade(moduleObj, request, tl, *args, **kwargs):
 # Just broke out this function to allow combined deadlines (see meets_any_deadline,
 # meets_all_deadlines functions below).  -Michael P, 6/23/2009
 def _checkDeadline_helper(method, extension, moduleObj, request, tl, *args, **kwargs):
-    if tl != 'learn' and tl != 'teach':
+    if tl != 'learn' and tl != 'teach' and tl != 'volunteer':
         return (True, None)
+    if "require_auth" in kwargs:
+        require_auth = kwargs["require_auth"]
+    else:
+        require_auth = True
     response = None
     canView = False
+    perm_name = {'learn':'Student','teach':'Teacher','volunteer':'Volunteer'}[tl]+extension
     if not_logged_in(request):
-        response = HttpResponseRedirect('%s?%s=%s' % (LOGIN_URL, REDIRECT_FIELD_NAME, quote(request.get_full_path())))
+        if not require_auth and Permission.valid_objects().filter(permission_type=perm_name, program=request.program, user__isnull=True).exists():
```

Comment:
(Per the below, you could just check `moduleObj.require_auth()` inline here instead of passing things around.